sdk/core/Azure.Core/src/Internal/SpanOfCharsExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    internal static class SpanOfCharsExtensions
+    {
+        public static void AppendString(this ref Span<char> span, ref int offset, string str)
+        {
+#if NET6_0_OR_GREATER
+            str.CopyTo(span[offset..]);
+#else
+            str.AsSpan().CopyTo(span.Slice(offset));
+#endif
+            offset += str.Length;
+        }
+
+        public static void AppendChar(this ref Span<char> span, ref int offset, char value)
+        {
+            span[offset] = value;
+            offset++;
+        }
+
+        public static void AppendLong(this ref Span<char> span, ref int offset, long value, string? format = null, IFormatProvider? provider = null)
```

Comment:
Curious: why add this overload, and not require the caller to pass a string?  Would we consider adding overloads for all types that can be formatted as string?

Reply:
> why add this overload, and not require the caller to pass a string?

Pass a string where?

> Would we consider adding overloads for all types that can be formatted as string?

Yeah, that sounds like a good idea. And I'll add tests for them as well.

Reply:
Whole type is removed in favor of `DefaultInterpolatedStringHandler`